scene/resources/animation.cpp

```diff
@@ -5594,7 +5594,10 @@ Variant Animation::array_to_string(const Variant p_value) {
 }
 
 Variant Animation::add_variant(const Variant &a, const Variant &b) {
-	if (a.get_type() != b.get_type() && !a.is_array()) {
+	if (a.is_num() && b.is_num()) {
+		const double result = (a.operator double()) + (b.operator double());
```

Comment:
It should not do the calculation here, but change the type of a to Variant::FLOAT to go through, and do the calculation in the switch-case block below. The same is applied to other parts.

Reply:
I now changed all these instances to call themselves with arguments converted to `double` (making sure to convert both `a` and `b` to prevent an infinite loop if `b` is `INT`), which essentially takes it to the switch with a new `a` (and possibly `b`), so this now is handled like the individual array elements.

I also changed the `INT` interpolation to go via `double` and `int64_t` which makes it work above `INT_MAX`, although not quite all the way to `INT64_MAX`, because that conversion isn't lossless. So the code in #81289 still does not work with this change, mostly because `subtract_variant(INT64_MAX, 0)` returns `INT64_MIN` instead of `INT64_MAX`. If we want to support that, we may need a dedicated code path for integer math, at least for subtraction.